src/client/metadataApiRetrieve.ts

```diff
@@ -185,9 +185,21 @@ export class MetadataApiRetrieve extends MetadataTransfer<
         fs.writeFileSync(zipFilePath, zipFileContents);
 
         if (this.options.unzip) {
-          const dir = await unzipper.Open.buffer(zipFileContents);
+          const zip = await JSZip.loadAsync(zipFileContents, { base64: true, createFolders: true });
           const extractPath = path.join(this.options.output, path.parse(name).name);
-          await dir.extract({ path: extractPath });
+          fs.mkdirSync(extractPath);
```

Comment:
checking my understanding: we need these mkdir/writeFile to be done sequentially to make sure the mkdir happens before any files that would go inside it?